freshmaker/errata.py

```diff
@@ -495,6 +495,28 @@ def get_cve_affected_rpm_nvrs(self, errata_id):
 
         return list(nvrs)
 
+    def get_cve_affected_build_nvrs(self, errata_id, only_modules=False):
+        """ Get module build NVRs which are affected by the CVEs in errata
+
+        :param errata_id: Errata advisory ID, e.g. 25713.
+        :type errata_id: str or int
+        :param only_modules: only return module builds NVR if True.
+        :type only_modules: bool
+        :return: a list of strings each of them is a build NVR
+        :rtype: list
+        """
+        data = self._errata_rest_get(f"/erratum/{errata_id}/builds_by_cve")
+        nvrs = set()
+        for data_by_product in data.values():
+            for product_data in data_by_product.values():
+                for build in product_data.get("builds", []):
+                    for build_info in build.values():
+                        if only_modules and not build_info.get("is_module"):
+                            continue
+                        nvrs.add(build_info.get("nvr"))
+
+        return list(nvrs)
```

Comment:
more python (and quick) way to do exactly the same thing...    
```
return list({build_info.get("nvr")
        for build_info in build.values()
        for build in product_data.get("builds", [])
        for product_data in data_by_product.values()
        for data_by_product in data.values()
        if not only_modules or build_info.get("is_module")})
```

Reply:
Thanks @mvalik for your review and suggestion!
There are two reasons to write this method in this way:
1. This method is imitating from previous one, we want to make them consistent. So no matter the looping style and  the return value, they are the same. If we change the return value from list to set, I am wondering if the user will be confused that similar method return different type of value.
2. Second,  I like one line style too, it is more pythonic. I think if the data structure is simple, I prefer this way, but this build data has several embedded layers, I think it is better to put them in loops because it is easy to debug and maintain.

Reply:
Performance is also a good point. Can we think about this potential performance issue in another patch, I'd like to deal it together with another method "get_cve_affected_rpm_nvrs", they are similar methods and  they all need to be improved.

Reply:
@qixiang @hluk what about your suggestions? Welcome to comment.

Reply:
@mvalik @chuazhan list comprehension is more pythonic, but when there are too many levels or the expression becomes too long, I'd prefer to avoid list comprehension for better code readability and maintainability.

Reply:
Since the performance impact is not that big in this case, we can keep it as loops.

Reply:
Cool. Thanks @mvalik